src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -1407,11 +1407,23 @@ impl<'a, 'tcx> Visitor<'tcx> for Checker<'a, 'tcx> {
                     }
                 }
                 ty::FnPtr(_) => {
+                    let unleash_miri = self
+                        .tcx
+                        .sess
+                        .opts
+                        .debugging_opts
+                        .unleash_the_miri_inside_of_you;
+                    let const_fn_ptr = self.tcx.features().const_fn_ptr;
                     if self.mode.requires_const_checking() {
-                        let mut err = self.tcx.sess.struct_span_err(
-                            self.span,
-                            &format!("function pointers are not allowed in const fn"));
-                        err.emit();
+                        if !(unleash_miri || const_fn_ptr) {
+                            emit_feature_err(
+                                &self.tcx.sess.parse_sess,
+                                sym::const_fn_ptr,
+                                self.span,
+                                GateIssue::Language,
+                                "function pointers in const fn are unstable",
```

Comment:
```suggestion
                                "function pointers in `const fn` are unstable",
```

Reply:
The other errors do not follow this style. cc @oli-obk

Reply:
The other errors are wrong then.

Reply:
Agreed, but I'd rather not touch them before https://github.com/rust-lang/rust/pull/64470 is merged

Reply:
I think that PR also enables function pointer calling behind unleash